src/hotspot/share/opto/macro.cpp

```diff
@@ -2571,6 +2577,7 @@ bool PhaseMacroExpand::expand_macro_nodes() {
     }
     assert(C->macro_count() == (old_macro_count - 1), "expansion must have deleted one node from macro list");
     if (C->failing())  return true;
+    C->print_method(PHASE_AFTER_MACRO_EXPANSION_ARRAYCOPY, 5, n);
```

Comment:
Should we rename this to something more generic since we are not only expanding array copy nodes but also locks and subtype checks? Maybe `PHASE_AFTER_MACRO_EXPANSION_STEP`?

Reply:
Sure, that sounds good. Maybe we should just have a single phase `PHASE_AFTER_MACRO_EXPANSION_STEP` to keep things simple? Even for the Opaque and LoopLimit node eliminations (as they are structurally part of the `expand_macro_nodes` function). We can refactor and move that eliminiation outside `expand_macro_nodes` in the RFE.

In summary, we then only have the phases:
```
  flags(BEFORE_MACRO_EXPANSION ,         "Before Macro Expansion") \
  flags(AFTER_MACRO_EXPANSION_STEP, "After Macro Expansion Step") \
  flags(AFTER_MACRO_EXPANSION,           "After Macro Expansion") \
```

Reply:
> Sure, that sounds good. Maybe we should just have a single phase `PHASE_AFTER_MACRO_EXPANSION_STEP` to keep things simple? Even for the Opaque and LoopLimit node eliminations (as they are structurally part of the `expand_macro_nodes` function). We can refactor and move that eliminiation outside `expand_macro_nodes` in the RFE.
> 
> In summary, we then only have the phases:
> 
> ```
>   flags(BEFORE_MACRO_EXPANSION ,         "Before Macro Expansion") \
>   flags(AFTER_MACRO_EXPANSION_STEP, "After Macro Expansion Step") \
>   flags(AFTER_MACRO_EXPANSION,           "After Macro Expansion") \
> ```

Sounds good to me.

Reply:
Good idea, let's go with that.